src/plugins/console/index.js

```diff
@@ -96,30 +111,64 @@ module.exports = function (kibana) {
         },
 
         pre: [
-          function filterUri(req, reply) {
-            const { uri } = req.query;
+          {
+            assign: 'proxyTarget',
+            method(req, reply) {
+              const rawUri = req.query.uri;
+              req.query.uri = null; // prevent accidental usage
+
+              const { protocol, hostname, port, pathname, search, hash } = parseUri(rawUri);
+              reply(formatUri({
+                protocol,
+                hostname,
+                port,
+                pathname: resolve(pathname), // resolve any path traversal
+                search,
+                hash
+              }));
+            }
+          },
 
-            if (!filters.some(re => re.test(uri))) {
+          function checkProxyTarget(req, reply) {
+            const uri = req.pre.proxyTarget;
+
+            if (!options.proxyTargets) return reply();
```

Comment:
This isn't actually an empty response, just how you finish a "pre" handler.

And it looks like Hapi automatically sends s 204 unless the status code is set explicitly